src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -933,6 +948,92 @@ class LoadDataSegment(BaseSegment):
     )
 
 
+# Data Retrieval Statements
+@spark3_dialect.segment()
+class ClusterByClauseSegment(BaseSegment):
+    """A `CLUSTER BY` clause from `SELECT` statement.
+
+    Equivalent to `DISTRIBUTE BY` and `SORT BY` in tandem.
+    This clause is mutually exclusive with SORT BY, ORDER BY and DISTRIBUTE BY.
+    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-clusterby.html
+    """
+
+    type = "cluster_by_clause"
+
+    match_grammar = StartsWith(
+        Sequence("CLUSTER", "BY"),
+        terminator=OneOf(
+            "LIMIT",
+            "HAVING",
+            # For window functions
+            "WINDOW",
+            Ref("FrameClauseUnitGrammar"),
+            "SEPARATOR",
+        ),
+    )
+
+    parse_grammar = Sequence(
+        "CLUSTER",
+        "BY",
+        Indent,
+        Delimited(
+            Sequence(
+                OneOf(
+                    Ref("ColumnReferenceSegment"),
+                    # Can `CLUSTER BY 1`
+                    Ref("NumericLiteralSegment"),
+                    # Can cluster by an expression
+                    Ref("ExpressionSegment"),
+                ),
+            ),
+            terminator=OneOf(
+                Ref.keyword("WINDOW"),
+                Ref.keyword("LIMIT"),
+                Ref("FrameClauseUnitGrammar"),
+            ),
+        ),
+        Dedent,
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class UnorderedSelectStatementSegment(BaseSegment):
+    """Enhance unordered `SELECT` statement to include Spark Clauses."""
+
+    type = "select_statement"
+
+    match_grammar = ansi_dialect.get_segment(
+        "UnorderedSelectStatementSegment"
+    ).match_grammar.copy()
+
+    parse_grammar = ansi_dialect.get_segment(
+        "UnorderedSelectStatementSegment"
+    ).parse_grammar.copy(
+        # TODO Insert: PIVOT and LATERAL VIEW clauses
+        remove=[Ref("OverlapsClauseSegment", optional=True)]
```

Comment:
Why is this in here? Is it part of this PR?

If it's intentional can you add a comment to the PR description and also to the code?

Reply:
It is intntional. I updated PR description and code